test/utils/web3_factory.ts

```diff
@@ -7,17 +7,21 @@ import ProviderEngine = require('web3-provider-engine');
 import RpcSubprovider = require('web3-provider-engine/subproviders/rpc');
 import * as Web3 from 'web3';
 import {constants} from './constants';
+import {EmptyWalletSubProvider} from '../../src/subproviders/empty_wallet_subprovider';
 
 export const web3Factory = {
-    create(): Web3 {
-        const provider = this.getRpcProvider();
+    create(hasAddresses: boolean = true): Web3 {
```

Comment:
I'd rather prefer `withAddresses`
`web3Factory.create(withAddresses)`, but it's subjective. Feel free to ignore